### PR TITLE
Redux react components

### PR DESCRIPTION
**Pull request by Alfredbis29**
**In a milestone of my Project I** 

First of all 
- make all states immutable in reducers!

Secondly, 

- change the default state in my book reducer from an empty array to an array with a few books. 
-  display them in my React components.
- Import `Redux` Provider and my store in the main component of my `React app`.
- Connect my app to the `Redux store`.
- list of books displays books from the `Redux store`.
- Use selectors to consume the state.

### CODE REVIEW

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @Alfredbis29, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f744b797263304e546c4563475764546870342f67697068792e676966

## Highlights  👏 
✅ Books can be added and removed 🔥 
✅ Good pull request description 👍 
✅ Linters are passing 👏 


## Required Changes ♻️
- [ ] Kindly arrange your categories reducer so that when a user clicks on the `check status` button, the text `under construction` should be displayed 👍 Right now the button does not have any functionality. Kindly fix that 😄 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Alfredbis29** [comment]
Hello @Kingstalux, I didn't see that requirement on this milestone of this project please check again all requirements, I'm looking forward to having your feedback. Please

**Comment by Kingstalux** [comment]
> Hello @Kingstalux, I didn't see that requirement on this milestone of this project please check again all requirements, I'm looking forward to having your feedback. Please
Annotation 2022-05-12 070849
That is it in the project requirements

**Comment by Alfredbis29** [comment]
Okay okay 
Thank you

File: src/components/Category.js [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import { checkStatus } from '../redux/categories/categories';
 
 const Category = () => {
   const dispatch = useDispatch();
   const checkStatusHandler = () => {
     dispatch(checkStatus());
   };
+  const categories = useSelector((state) => state.categories);
```
Comment (AnselemOdims) [inline src/components/Category.js:10]: - [ ][OPTIONAL] It is usually good practice to put your hooks before any handler functions

**Review by AnselemOdims — APPROVED** [review]
### Hi  @Alfredbis29 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**